sympy/tensor/tests/test_indexed.py

```diff
@@ -422,6 +422,13 @@ def test_issue_12780():
     raises(TypeError, lambda: i.subs(n, 1.5))
 
 
+def test_issue_18604():
+    m = symbols("m")
+    assert Idx("i", m).name == 'i'
+    assert Idx("i", m).lower == 0
+    assert Idx("i", m).upper == m - 1
```

Comment:
I'm afraid this can end up with incorrect notations.
What happens if `m` is a negative number, a real number, or a complex number? 
Should lower, upper be bounded with `0` and `m-1` even for those cases?

Reply:
For complex number it raises an error, but for negative and real it does'nt

Reply:
I think what is here now is ok. The range can be real but if a non-integer is substituted for it then an error will be raised. We just don't want an error raised if range is a symbol that could be an integer. We only want an error when the range is definitely invalid.

Reply:
Thanks @smichr. Does it look good now?